ror-tools-core/src/main/scala/tech/beshu/ror/tools/core/patches/internal/filePatchers/JarManifestModifier.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.tools.core.patches.internal.filePatchers
+
+import better.files.*
+import tech.beshu.ror.tools.core.utils.EsDirectory
+
+import java.util.UUID
+import java.util.jar.{JarFile, JarOutputStream}
+import scala.jdk.CollectionConverters.IteratorHasAsScala
+import scala.util.Using
+
+object JarManifestModifier {
+
+  private val patchedByRorVersionPropertyName = "Patched-By-Ror-Version"
+
+  def addPatchedByRorVersionProperty(file: File, rorVersion: String): Unit = {
+    Using(new JarFile(file.toJava)) { jarFile =>
+      val tempJarFile = File(s"temp-${UUID.randomUUID()}.jar")
+      val manifest = jarFile.getManifest
+      manifest.getMainAttributes.putValue(patchedByRorVersionPropertyName, rorVersion)
+      Using(new JarOutputStream(tempJarFile.newOutputStream.buffered, manifest)) { jarOutput =>
```

Comment:
maybe we can extract this to a well-named private method to increase readability? WDYT?